src/benchmarks/gc/GC.Infrastructure/GC.Infrastructure/Commands/ASPNetBenchmarks/AspNetBenchmarksCommand.cs

```diff
@@ -42,11 +42,167 @@ public override int Execute([NotNull] CommandContext context, [NotNull] AspNetBe
 
             ConfigurationChecker.VerifyFile(settings.ConfigurationPath, nameof(AspNetBenchmarksCommand));
             ASPNetBenchmarksConfiguration configuration = ASPNetBenchmarksConfigurationParser.Parse(settings.ConfigurationPath);
+
+            // Before running the ASP.NET benchmarks, execute a few checks:
+            // Check 1. If you are connected to corpnet. These tests only run if you are connected to corp-net and therefore, preemptively, check this.
+            // Check 2. The host machines reboot between 12:00 AM - 12:08 AM PST. Check if we are running during that time, and if so, delay running the infrastructure.
+
+            // Check 1.
+            if (!Directory.Exists(@"\\clrmain\tools\"))
+            {
+                // Not connected to corpnet if that folder is unavailable.
+                AnsiConsole.MarkupLine($"[red bold] Not connected to corpnet. Ensure you are connected before proceeding with the ASP.NET Benchmarks [/]");
+                return -1;
+            }
+
+            // Check 2.
+            SleepUntilHostsHaveRestarted();
```

Comment:
Is it correct that this check is at the beginning of the overall run?  This check is probably best at the beginning of each benchmark (in fact, each attempt at each benchmark in the case of a retry)

Reply:
100% - adding now.